docs/developer.md

```diff
@@ -31,14 +31,21 @@ guidance on pip installation (e.g., for GPU and TPU support).
 
 ### Building `jaxlib` from source
 
+```{warning}
+While it should typically be possible to compile `jaxlib` from source using
+most modern compilers, the builds are only tested using `clang`. Pull requests
+are welcomed to improve support for different toolchains, but they are not
+actively supported.
+```
+
 To build `jaxlib` from source, you must also install some prerequisites:
 
 - a C++ compiler (g++, clang, or MSVC)
 
   On Ubuntu or Debian you can install the necessary prerequisites with:
 
   ```
-  sudo apt install g++ python python3-dev
+  sudo apt install clang python python3-dev
```

Comment:
It's actually not just `clang` but a sufficiently recent clang. I think clang 18 is our current version.

On Ubuntu 22.04, the clang is too old, so you actually need to do something more like this:
```
wget -qO- https://apt.llvm.org/llvm-snapshot.gpg.key | sudo tee /etc/apt/trusted.gpg.d/apt.llvm.org.asc
echo deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-18 main | sudo tee /etc/apt/sources.list
sudo apt install clang-18 libstdc++12-dev
```

You may wish to mention that.

Reply:
Good suggestion - thanks. I added a comment about the required version and a link to https://apt.llvm.org because that seemed better than maintaining our own version of the instructions, but I'm happy to add more specific details here if you think that would be better!